File: frontend/src/services/OrderItemService.js
import BaseService from 'src/services/base/BaseService'

export default class OrderItemService extends BaseService {
  constructor() {
    super("orderItem");
  }

  async findAll() {
    return await this.GET('');
  }

  async findById(id) {
    return await this.GET(`/id/${id}`);
  }

  async findAllOptimized() {
    return await this.GET(`/optimized`);
  }

  async findPaginated(page = 0, size = 10) {
    return await this.GET(`/paginated?page=${page}&size=${size}`);
  }

  async create(body) {
    return await this.POST('', body);
  }

  async createList(body) {
    return await this.POST(`/all`, body);
  }

  async update(body) {
    return await this.PUT('', body);
  }

  async updateList(body) {
    return await this.PUT(`/all`, body);
  }

  async delete(body) {
    return await this.DELETE('', body);
  }

  async deleteAll(body) {
    return await this.DELETE('', body);
  }

  async deleteById(id) {
    return await this.DELETE(`/id/${id}`);
  }

  async deleteByOrderId(id) {
    return await this.DELETE(`/order/${id}`);
  }

  async findByOrderId(id) {
    return await this.GET(`/order/${id}`);
  }
  async deleteByMenuItemId(id) {
    return await this.DELETE(`/menuitem/${id}`);
  }

  async findByMenuItemId(id) {
    return await this.GET(`/menuitem/${id}`);
  }
}

File: backend/src/main/java/org/bshg/demo/services/facade/OrderItemService.java
package org.bshg.demo.services.facade;
import org.bshg.demo.entity.core.OrderItem;
import org.bshg.demo.entity.core.Order;
import org.bshg.demo.entity.core.MenuItem;
import org.bshg.demo.zutils.pagination.Pagination;
import java.util.List;
public interface OrderItemService {
OrderItem findById(Long id);
List<OrderItem> findAllOptimized();
List<OrderItem> findAll();
Pagination<OrderItem> findPaginated(int page, int size);
OrderItem create(OrderItem item);
List<OrderItem> create(List<OrderItem> item);
OrderItem update(OrderItem item);
List<OrderItem> update(List<OrderItem> item);
void deleteById(Long id);
void delete(OrderItem item);
void delete(List<OrderItem> items);
void deleteByIdIn(List<Long> ids);
int deleteByOrderId(Long id);
List<OrderItem> findByOrderId(Long id);
int deleteByMenuItemId(Long id);
List<OrderItem> findByMenuItemId(Long id);
}